web/pwa/src/features/recorder/FieldEditor.tsx

```diff
@@ -71,6 +71,61 @@ export const DateFieldEditor: FC<FieldEditorProps> = props => {
     </div>
 }
 
+export const WeekFieldEditor: FC<FieldEditorProps> = props => {
+    const {field, value, setValue} = props
+    const expectedLength = 8;
+
+    const [localValue, setLocalValue] = useState(value != null ? getFormattedWeekStringFromDate(value) : "")
+
+    const isValidLength = () => localValue.length === expectedLength;
+
+    function getDateFromWeekN(w: number, y: number) {
+        const d = (1 + (w - 1) * 7); // 1st of January + 7 days for each week
+        return new Date(y, 0, d);
+    }
+
+    function getWeekFromDate(date: Date) {
+        const oneJan = new Date(date.getFullYear(), 0, 1);
+        // @ts-ignore
+        const numberOfDays = Math.floor((date - oneJan) / (24 * 60 * 60 * 1000));
+        return Math.ceil((date.getDay() + 1 + numberOfDays) / 7);
+    }
+
+    function getFormattedWeekStringFromDate(date: Date) {
+        const w = getWeekFromDate(date);
+        const y = date.getFullYear();
+        return `${y}-W${w}`
+    }
+
+    function isValid(s: string) {
+        const valid = /^(?:19|20|21)\d{2}-W[0-5]\d$/
+        return valid.test(s) && +s.slice(6) <= 52;
+    }
```

Comment:
I think this implementation will not be sufficient.. ISO dates are quite complicated!
Perhaps we could use dateFns ? https://www.npmjs.com/package/date-fns